README.md

```diff
@@ -105,6 +114,35 @@ In the `"Serilog.WriteTo"` array, add an entry for `DatadogLogs`. An example is
 }
 ```
 
+## Inner workings
+The logger asynchronously emits events in batches that are set to run every 2 seconds. 10 retries are made, spaced out by an exponential retry using the following formula:  
+`int backoff = (int)Math.Min(Math.Pow(2, retry), MaxBackoff);`  
+
+`MaxBackOff` is set to 30. 
+
+If the batch fails due to an exception (for example invalid url) the exception is swallowed and it keeps trying until there are no retries left. If none of the attempts have gone through it will log to the SelfLog (seriLog feature) using `SelfLog.WriteLine`. 
+
+Call SelfLog.Enable() at program startup to enable the log.  
+
+Example using Debug.WriteLine:
+
+```
+// assume logger setup from earlier example
+
+Serilog.Debugging.SelfLog.Enable(msg => Debug.WriteLine(msg));
+
+log.Information("New ping");
+// output
+
+2020-06-30T18:56:07.6409072Z Could not send payload to Datadog: [{"Timestamp":"2020-06-30T20:56:07.2296983+02:00","level":"Information","MessageTemplate":"New ping","message":"New ping","ddsource":"csharp","service":"myService"}]
+```
+The exception is not logged.
+
+#### Important note 
+due to the periodic batching and exponential backoff retries *an entry might get lost if the application closes or restarts before the batch has been emitted, or retries attempts have not been exhausted.*  
+
+If you want to test the logging in a console application make sure to add a delay that allows enough time to process the request. This can be done by using , for example, Thread.Sleep(time) or Task.Delay(time). Blocking the thread by using Thread.Sleep is generally not recommended, but can be used if you just want to verify that the configuration works.
```

Comment:
```suggestion
If you want to test the logging in a console application, make sure to add a delay that allows enough time to process the request. This can be done by using, for example, Thread.Sleep(time) or Task.Delay(time). Blocking the thread by using Thread.Sleep is generally not recommended, but can be used if you just want to verify that the configuration works.
```